File: refers/_decoded/extract_features.py
# -*- coding: utf-8 -*-
import re
import os

base = r'd:\WorkSpace\Projects\TkSwarm-Rebuild\refers\_decoded'

for fn in sorted(os.listdir(base)):
    if not fn.endswith('.html'):
        continue
    path = os.path.join(base, fn)
    with open(path, 'r', encoding='utf-8') as f:
        c = f.read()

    zh = sorted(set(re.findall(
        r'[\u4e00-\u9fff][\u4e00-\u9fff0-9A-Za-z_\-\s/（）()：:，,、·.#]{1,50}', c)))

    apis = re.findall(r'''['"`]/api/[a-zA-Z0-9_/\-${}.]+''', c)
    apis = sorted(set(a[1:] if a[0] in "'\"`" else a for a in apis))

    # buttons / menu items from el-button / el-menu-item text
    buttons = re.findall(
        r'<(?:el-button|el-menu-item|el-tab-pane|el-dropdown-item|button)[^>]*>\s*([^<]{1,40})\s*<',
        c)
    buttons += re.findall(r'slot="title">\s*([^<]{1,40})\s*<', c)
    buttons += re.findall(r'<span[^>]*>\s*([\u4e00-\u9fff][^<]{0,40})\s*</span>', c)
    buttons = sorted(set(b.strip() for b in buttons if b.strip()))

    # dialog titles
    dialogs = re.findall(r':title="([^"]+)"', c)
    dialogs += re.findall(r'title="([^"]+)"', c)
    dialogs = sorted(set(dialogs))

    # websocket / message types
    ws_types = sorted(set(re.findall(
        r'''(?:type|msgType|messageType|event)\s*[:=]\s*['"]([a-zA-Z0-9_]+)['"]''', c)))

    # status maps in JS comments or objects
    status_blocks = re.findall(
        r'(?:statusMap|statusText|statusLabel|taskStatus|loginStatus|friendStatus)[\s\S]{0,400}',
        c)

    # form / settings keys
    form_keys = sorted(set(re.findall(
        r'''(?:this\.)?([a-zA-Z][a-zA-Z0-9]*(?:Form|Settings|Config|Dialog|Visible|List|Map|Options))\b''',
        c)))

    # el-option labels
    options = re.findall(
        r'''<(?:el-option|el-radio|el-checkbox)[^>]*(?:label|title)=["']([^"']+)["']''', c)
    options += re.findall(
        r'''<(?:el-option|el-radio)[^>]*>\s*([^<]{1,40})\s*<''', c)
    options = sorted(set(o.strip() for o in options if o.strip()))

    # case 'xxx' in switch for WS
    cases = sorted(set(re.findall(r'''case\s+['"]([a-zA-Z0-9_\-]+)['"]''', c)))

    out_path = os.path.join(base, fn.replace('.html', '_extract.txt'))
    with open(out_path, 'w', encoding='utf-8') as o:
        o.write('=== TITLE ===\n')
        m = re.search(r'<title>([^<]+)</title>', c)
        o.write((m.group(1) if m else '?') + '\n\n')

        o.write('=== APIS (%d) ===\n' % len(apis))
        for a in apis:
            o.write(a + '\n')

        o.write('\n=== DIALOG TITLES ===\n')
        for d in dialogs:
            o.write(d + '\n')

        o.write('\n=== BUTTONS/MENUS ===\n')
        for b in buttons:
            o.write(b + '\n')

        o.write('\n=== OPTIONS/RADIOS ===\n')
        for opt in options:
            o.write(opt + '\n')

        o.write('\n=== FORM/STATE KEYS ===\n')
        for k in form_keys:
            o.write(k + '\n')

        o.write('\n=== SWITCH CASES / WS TYPES ===\n')
        for x in sorted(set(ws_types + cases)):
            o.write(x + '\n')

        o.write('\n=== STATUS BLOCKS ===\n')
        for s in status_blocks[:20]:
            o.write(s[:300].replace('\n', ' ') + '\n---\n')

        o.write('\n=== ZH STRINGS (%d) ===\n' % len(zh))
        for z in zh:
            o.write(z.strip() + '\n')

    print('Wrote', out_path, 'zh=', len(zh), 'apis=', len(apis), 'btns=', len(buttons))

File: refers/_decoded/extract_features.js
const fs = require('fs');
const path = require('path');
const base = __dirname;

for (const fn of fs.readdirSync(base).filter(f => f.endsWith('.html') && f.startsWith('page_'))) {
  const c = fs.readFileSync(path.join(base, fn), 'utf8');
  const zh = [...new Set(c.match(/[\u4e00-\u9fff][\u4e00-\u9fff0-9A-Za-z_\-\s/（）()：:，,、·.#]{1,50}/g) || [])].sort();
  const apis = [...new Set((c.match(/['"`]\/api\/[a-zA-Z0-9_/\-${}.]+/g) || []).map(a => a.slice(1)))].sort();
  const dialogs = [...new Set([
    ...(c.match(/title="[^"]+"/g) || []).map(t => t.slice(7, -1)),
    ...(c.match(/:title="[^"]+"/g) || []).map(t => t.slice(8, -1)),
  ])].sort();
  const cases = [...new Set((c.match(/case\s+['"]([a-zA-Z0-9_\-]+)['"]/g) || []).map(x => x.replace(/case\s+['"]|['"]/g, '')))].sort();
  const forms = [...new Set(c.match(/\b[a-zA-Z][a-zA-Z0-9]*(?:Form|Settings|Config|Dialog|Visible|Options|Map|List)\b/g) || [])].sort();
  const elDialogs = [...new Set((c.match(/<el-dialog[^>]*title="([^"]+)"/g) || []).map(m => (m.match(/title="([^"]+)"/) || [])[1]).filter(Boolean))];
  const tabs = [...new Set((c.match(/<el-tab-pane[^>]*label="([^"]+)"/g) || []).map(m => (m.match(/label="([^"]+)"/) || [])[1]).filter(Boolean))];
  const dropdowns = [...new Set((c.match(/<el-dropdown-item[^>]*>[\s\S]*?<\/el-dropdown-item>/g) || [])
    .map(m => m.replace(/<[^>]+>/g, ' ').replace(/\s+/g, ' ').trim()).filter(Boolean))];
  const buttons = [...new Set((c.match(/<el-button[^>]*>[\s\S]*?<\/el-button>/g) || [])
    .map(m => m.replace(/<[^>]+>/g, ' ').replace(/\s+/g, ' ').trim()).filter(x => x.length > 0 && x.length < 40))];

  const out = [
    '=== TITLE ===',
    (c.match(/<title>([^<]+)<\/title>/) || [, '?'])[1],
    '',
    `=== APIS (${apis.length}) ===`,
    ...apis,
    '',
    '=== EL-DIALOGS ===',
    ...elDialogs,
    '',
    '=== TABS ===',
    ...tabs,
    '',
    '=== DIALOG/TITLE ATTRS ===',
    ...dialogs.slice(0, 80),
    '',
    '=== DROPDOWN ITEMS ===',
    ...dropdowns.slice(0, 80),
    '',
    '=== BUTTON TEXTS (sample) ===',
    ...buttons.slice(0, 120),
    '',
    '=== FORM/STATE KEYS ===',
    ...forms,
    '',
    '=== SWITCH CASES ===',
    ...cases,
    '',
    `=== ZH (${zh.length}) ===`,
    ...zh,
  ].join('\n');

  fs.writeFileSync(path.join(base, fn.replace('.html', '_extract.txt')), out, 'utf8');
  console.log('Wrote', fn, 'zh=', zh.length, 'apis=', apis.length);
}
